transport.go

```diff
@@ -158,6 +166,11 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	_ = res.Body.Close()
 	// save the challenge for future use
 	if err := t.save(res); err != nil {
+		// corner case for weird implementations: if no WWW-Authenticate header was found
+		if errors.Is(err, ErrNoWWWAuthenticateHeaders) {
+			// forget the challenge if it was present
+			t.forgetChallenge(req)
+		}
 		return nil, err
```

Comment:
I'm wondering if it would be more correct to simply return the `http.Response` and `nil` error in the case of no challenge being present.

Reply:
Let's leave this as-is for now.